src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -72,26 +72,35 @@ protected ClusterBlockException checkBlock(GetSnapshotsRequest request, ClusterS
 
     @Override
     protected void masterOperation(final GetSnapshotsRequest request, ClusterState state, final ActionListener<GetSnapshotsResponse> listener) throws ElasticsearchException {
-        SnapshotId[] snapshotIds = new SnapshotId[request.snapshots().length];
-        for (int i = 0; i < snapshotIds.length; i++) {
-            snapshotIds[i] = new SnapshotId(request.repository(), request.snapshots()[i]);
-        }
-
         try {
             ImmutableList.Builder<SnapshotInfo> snapshotInfoBuilder = ImmutableList.builder();
-            if (snapshotIds.length > 0) {
-                for (SnapshotId snapshotId : snapshotIds) {
-                    snapshotInfoBuilder.add(new SnapshotInfo(snapshotsService.snapshot(snapshotId)));
-                }
-            } else {
+            if (isAllSnapshots(request.snapshots())) {
                 ImmutableList<Snapshot> snapshots = snapshotsService.snapshots(request.repository());
                 for (Snapshot snapshot : snapshots) {
                     snapshotInfoBuilder.add(new SnapshotInfo(snapshot));
                 }
+            } else if (isCurrentSnapshots(request.snapshots())) {
+                ImmutableList<Snapshot> snapshots = snapshotsService.currentSnapshots(request.repository());
+                for (Snapshot snapshot : snapshots) {
+                    snapshotInfoBuilder.add(new SnapshotInfo(snapshot));
+                }
+            } else {
+                for (int i = 0; i < request.snapshots().length; i++) {
+                    SnapshotId snapshotId = new SnapshotId(request.repository(), request.snapshots()[i]);
+                    snapshotInfoBuilder.add(new SnapshotInfo(snapshotsService.snapshot(snapshotId)));
+                }
             }
             listener.onResponse(new GetSnapshotsResponse(snapshotInfoBuilder.build()));
         } catch (Throwable t) {
             listener.onFailure(t);
         }
     }
+
+    private boolean isAllSnapshots(String[] snapshots) {
+        return (snapshots.length == 0) || (snapshots.length == 1 && "_all".equalsIgnoreCase(snapshots[0]));
+    }
+
+    private boolean isCurrentSnapshots(String[] snapshots) {
+        return (snapshots.length == 1 && "_current".equalsIgnoreCase(snapshots[0]));
```

Comment:
Can you make `"_all"` and `"_current"` public static strings somewhere appropriate so they can be used by the Java API?